core/src/main/java/org/wso2/testgrid/core/command/GenerateTestPlanCommand.java

```diff
@@ -628,4 +631,35 @@ private Path getTestGridYamlLocation(String directory) {
             return defaultYamlPath;
         }
     }
+
+    /**
+     * Validate the testgridYaml. It must contain at least one infra provisioner, and one scenario.
+     * @param testgridYaml TestgridYaml object
+     * @return True or False, based on the validity of the testgridYaml
+     */
+    private boolean validateTestgridYaml(TestgridYaml testgridYaml) {
+        Boolean isValidTestgridYaml = true;
+        InfrastructureConfig infrastructureConfig = testgridYaml.getInfrastructureConfig();
+        ScenarioConfig scenarioConfig = testgridYaml.getScenarioConfig();
+        if (infrastructureConfig != null) {
+            if (infrastructureConfig.getProvisioners().isEmpty()) {
+                logger.error("testgrid.yaml doesn't contain at least single infra provisioner, Invalid testgrid.yaml");
+                isValidTestgridYaml = false;
+            }
+        } else {
+            logger.error("testgrid.yaml doesn't have defined the infra configuration, Invalid testgrid.yaml");
```

Comment:
The decision of logging this is upto the caller of #validateTestgridYaml method. The validateTestgridYaml method's sole purpose is to check the validity and return true/false imo.

Reply:
This is fixed 9be59e94046fe2feb8fe36c3e1d77913ba1669a7